### ISSUE TITLE
Audio stopped working on Linux Mint 22

### ISSUE THREAD

**Issue by defensem3ch** [jkotlinski/lsdpatch#100]
It seems that I am not able to hear any audio output from the kit editor, possibly due to the patcher selecting an incorrect default audio output? I am still testing pipewire audio setup in Linux Mint, but it might be nice to consider if there is an easy way to select an audio device, possibly via commandline argument.

**Comment by jkotlinski** [comment]
Hmm isn’t that functionality supposed to be built into the operating system
itself?

lör 21 sep. 2024 kl. 14:30 skrev DEFENSE MECHANISM ***@***.***

**Comment by jkotlinski** [comment]
…by that i mean… if i have windows running, i can click a speaker icon and
select the output…
Is it different in Linux?

lör 21 sep. 2024 kl. 18:34 skrev Johan Kotlinski ***@***.***>:

**Comment by defensem3ch** [comment]
Yes, but the java app seems perhaps to ignore the default audio device. I've spent some time researching the issue but unfortunately haven't stumbled upon anything helpful yet.

**Comment by jhob101** [comment]
Not sure if this is related, but wav files exported from Reaper only play the first 1 second in kit manager, but the whole loop is actually saved to the kit.

Other wav files seem fine, not quite sure what the difference would be.

**Comment by jkotlinski** [comment]
That seems weird. Could you attach such a wav file here to this issue and I
can have a look?

mån 23 sep. 2024 kl. 11:10 skrev John Hobson ***@***.***>:

**Comment by jhob101** [comment]
I think I just figured it out.  It's ok if exported as 16bit PCM wav, it's 24bit PCM wavs that it just plays first second.

So there is as solution, probably not a priority to fix, if a fix is even needed.

Links to download the 2 wavs:
24 bit: https://i.damselflycreative.com/media/16bit.wav
16 bit: https://i.damselflycreative.com/media/24bit.wav

**Comment by jhob101** [comment]
Good to know.  16 bit is fine anyway, not like you need especially high quality source audio!

I'm using pipewire, so might be something related to that.

**Comment by defensem3ch** [comment]
I was able to replace some of the code in Sound.java to use the "pulse" audio device and this is working for me. In the future it would be more ideal to implement this as a menu item, I may work on this and provide a pull request if I ever have enough free time.

```java
// Get the specific mixer (audio device) containing "M4" in its name
        Mixer mixer = getMixerByName("pulse");
        if (mixer == null) {
            throw new LineUnavailableException("No audio device found containing 'pulse' in its name.");
        }

        // Get the Clip from the selected mixer
        Clip clip = (Clip) mixer.getLine(new DataLine.Info(Clip.class, new AudioFormat(PLAYBACK_RATE, 16, 1, true, false)));
```

**Comment by PeBoVision** [comment]
I've installed both ZorinOS/Mint/Ubuntu on a variety of laptops with zero sound issues. I installled Bazzite/Fedora on my game system and ditto...

But when I intallled Mint 22 on my Intel  8th generation NUC, that includes both 5.1 audio via the Sony Vega 6/HDMI, and a Realtek ALC700 outputing to optical out, all hell broke loose. My aging receiver does not play welll with decoding AC3 and DTS over HDMI, so I rely on Optical out for sound.

I shut off the AMD audio off in pavucontrol, which forced the Realtek audio as the default, and I got sound, but it is AWFUL, it cuts out randomly, and eventually cuts out completely only reparable through a reboot (not even resseting pipewire works without a reboot.) And when it does work it is distorted at ANY volume.

This is becoming more frustrating than my geriatric  nerves can take, and I am seriouslly thinking of just installing Win11 and tolerating the loss of privacy...so if someone can offer a solution to optical out systems, and save me from MS hell, it would be welcome.

(also, I set the advanced output control to include AC3 and DTS, which the chip supports butevery thing is coming through as PCH, so not AC3 or DTS 5.1 which Windows supports without issue..)

Since I have never had sound issues until now, this is an area of Ubuntu I am unfamiliar with, so speak as you would to a child...I'm not yet senile, but close.

**Comment by jkotlinski** [comment]
Hi! Sorry to hear that you are having problems. I cannot do much to help,
but I will leave this issue open in case someone else can chime in.

tis 23 sep. 2025 kl. 01:40 skrev PeBoVision ***@***.***>:
